tempodb/blocklist/poller.go

```diff
@@ -196,11 +201,18 @@ func (p *Poller) pollTenantAndCreateIndex(
 	// are we a tenant index builder?
 	builder := p.tenantIndexBuilder(tenantID)
 	span.SetTag("tenant_index_builder", builder)
+
 	if !builder {
 		metricTenantIndexBuilder.WithLabelValues(tenantID).Set(0)
 
 		i, err := p.reader.TenantIndex(derivedCtx, tenantID)
 		err = p.tenantIndexPollError(i, err)
+
+		if errors.Is(err, backend.ErrDoesNotExist) {
```

Comment:
so the negative here is if we manually delete a tenant index to force recreation then only the assigned tenant index builders will attempt to recreate. which means that certain queriers will be unaware of any blocks in the backend for a tenant until it gets recreated (multiple minutes).

instead we could
1) log info/debug "no tenant index found. falling through to polling" 
2) falling through to the blocklist poll below
3) if the full blocklist list call returns an empty list log a warning. "no blocks found for tenant that should exist"
4) raise the metric `tempod_blocklist_poll_errors_total` to build alerts off of/warn the operator to clean up partial blocks

wdyt?

Reply:
I think we want to avoid polling in the case of a missing index.  The only signal that we have currently that we wish to delete the tenant, is to delete the index.  If you wanted to force a recreation of the index, you could corrupt the index, which would cause the poller to poll and recreate it.